### DISCUSSION TITLE
What are the compatible versions of CUDA and Visual Studio for building onnxruntime-win-x64-gpu-1.16.3 on Windows?

### DISCUSSION THREAD

**Discussion by shiwenloong** [microsoft/onnxruntime#19237] [EP Q&A] [answered]
I have encountered compatibility issues while attempting to compile ONNX Runtime version 1.16.3 with GPU support on a Windows environment. My local setup includes CUDA 11.3 and Visual Studio 2019.

I am seeking information on the recommended environment for compiling ONNX Runtime 1.16.3 with GPU support. Specifically, I am interested in knowing the compatible versions of CUDA and Visual Studio that are recommended for this particular release.

**Comment by carzh** [comment, marked as answer]
Here is the [CUDA version compatibility matrix](https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html#requirements) -- it looks like onnxruntime 1.16 can support CUDA 11.6 to 11.8. I believe there are no Visual Studio requirements for building onnxruntime with CUDA. [This document](https://onnxruntime.ai/docs/build/eps.html#cuda) has more information about building onnxruntime with CUDA :)
